### PR TITLE
Add machine time calculation functions for MF_CCVM

### PR DESCRIPTION
**Pull request by KatieO4t1qbit**
This commit adds an option to calculate the machine_time using MF_CCVM. This option can only be used if the solver is MFSolver.

Closes https://github.com/1QB-Information-Technologies/ccvm/issues/158

Future commits (other PRs) will add the calculations for other optics solvers.

Testing
You can verify this feature by:

* running the new unit tests
* customizing the solver type and/or the machine parameter of machine_time() in the examples/ccvm_boxqp_plot.py file and then running it

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

File: ccvm_simulators/solvers/mf_solver.py [hunk 1]
```diff
@@ -398,6 +398,65 @@ def _optics_machine_energy_callable(
...
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average average time spent by the solver on a single instance.
+            """
+            self._validate_machine_time_dataframe_columns(dataframe)
+
+            # TODO before merge: Ask Farhad if using mean was correct here
+            iterations = np.mean(dataframe["iterations"].values)
```
Comment (KatieO4t1qbit) [inline ccvm_simulators/solvers/mf_solver.py:443]: @FarhadK-1QBit I assume using `np.mean()` here was intended, like Estella's PRs did, but I just want to confirm with you!

File: ccvm_simulators/solvers/mf_solver.py [hunk 2]
```diff
@@ -398,6 +398,65 @@ def _optics_machine_energy_callable(
...
+            machine_parameters (dict, optional): Parameters of the optical MF-CCVM
+                machine. Defaults to None.
+
+        Raises:
+            ValueError: when the given machine parameters are not valid.
+            ValueError: when the given dataframe does not contain the required columns.
+
+        Returns:
+            Callable: A callable function that takes in a dataframe and problem size and
+                returns the average average time spent by the solver on a single instance.
```
Comment (KatieO4t1qbit) [inline ccvm_simulators/solvers/mf_solver.py:416]: Average wasn't average enough 😂  (I made this mistake a few places)
```suggestion
                returns the average time spent by the solver on a single instance.
```

File: ccvm_simulators/solvers/mf_solver.py (hunk 1 above)
Comment (FarhadK-1QBit) [reply ccvm_simulators/solvers/mf_solver.py:443]: That's right, Katie!

**Review by FarhadK-1QBit — APPROVED** [review]
LGTM, @KatieO4t1qbit. Thank you!

File: ccvm_simulators/solvers/mf_solver.py [hunk 3]
```diff
@@ -398,6 +398,65 @@ def _optics_machine_energy_callable(
...
+            Raises:
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average average time spent by the solver on a single instance.
+            """
+            self._validate_machine_time_dataframe_columns(dataframe)
+
+            # TODO before merge: Ask Farhad if using mean was correct here
```
Comment (ChanWooY-1QBit) [inline ccvm_simulators/solvers/mf_solver.py:442]: Just make sure to remove this before merging.

**Review by ChanWooY-1QBit — APPROVED** [review]
Looking good! Please address my comment before merging. Thanks!

**Review by Estella-Wang-1qbit — APPROVED** [review]
Thanks Katie. Looks good to me! 💯

**Pull request merged**